controller/snapshot_controller.go

```diff
@@ -211,7 +213,51 @@ func filterSnapshotsForEngineEnqueuing(oldEngine, curEngine *longhorn.Engine, sn
 	return targetSnapshots
 }
 
-func (sc *SnapshotController) enqueueVolumeChange(obj interface{}) {
+func (sc *SnapshotController) enqueueVolumeChange(old, new interface{}) {
+	oldVol, ok := old.(*longhorn.Volume)
+	if !ok {
+		utilruntime.HandleError(fmt.Errorf("received unexpected obj: %#v", old))
+		return
+	}
+	newVol, ok := new.(*longhorn.Volume)
```

Comment:
NIT: Would you add a comment that explains when newVol should be handled by SnapshotController?

Reply:
Added. Thanks. Please review it again to ensure that nothing needs to be modified.